MaterialLib/SolidModels/CreateConstitutiveRelation.cpp

```diff
@@ -0,0 +1,84 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *  \file   CreateConstitutiveRelation.cpp
+ *  Created on July 10, 2018, 12:09 PM
+ */
+
+#include "CreateConstitutiveRelation.h"
+#include "CreateEhlers.h"
+#include "CreateLinearElasticIsotropic.h"
+#include "CreateLubby2.h"
+
+#include "MechanicsBase.h"
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace MaterialLib
+{
+namespace Solids
+{
+template <int DisplacementDim>
+std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
+createConstitutiveRelation(
+    std::vector<std::unique_ptr<ProcessLib::ParameterBase>> const& parameters,
+    BaseLib::ConfigTree const& config)
+{
+    // Constitutive relation.
+    // read type;
+    auto const constitutive_relation_config =
+        //! \ogs_file_param{material__solid__constitutive_relation}
+        config.getConfigSubtree("constitutive_relation");
+
+    auto const type =
+        //! \ogs_file_param{material__solid__constitutive_relation__type}
+        constitutive_relation_config.peekConfigParameter<std::string>("type");
+
+    std::unique_ptr<MaterialLib::Solids::MechanicsBase<DisplacementDim>>
+        material = nullptr;
+    if (type == "Ehlers")
+    {
+        material = MaterialLib::Solids::Ehlers::createEhlers<DisplacementDim>(
```

Comment:
Minor: You could just `return MaterialLib::Solids::Ehlers::createEhlers<DisplacementDim>(...`

Reply:
Changed according by amending the second commit.